controller/src/controller.rs

```diff
@@ -927,17 +929,16 @@ impl Controller {
         let response = self.rpc(request).await?;
         match response.message.body {
             MessageBody::SpResponse(
-                st @ SpResponse::Status {
+                st @ SpResponse::StatusV2 {
                     modules,
                     failed_modules,
                 },
             ) => {
                 let data = response.data.expect("Existence of data checked earlier");
                 let (data, error_data) = data.split_at(st.expected_data_len().unwrap());
                 let status = data
-                    .iter()
-                    .copied()
-                    .map(|x| Status::from_bits(x).unwrap())
+                    .chunks_exact(4)
+                    .map(|x| hubpack::deserialize(x).unwrap().0)
```

Comment:
Here we are using `hubpack::deserialize`, but in `impl SpResponse` we compute the size based on `size_of`. I think this is all self-consistent, because we make `SerializedSize::MAX_SIZE` equal to `size_of`, it's just confusing. I think I'd suggest we use `from_bits` here as well; or remove the comment about using `from_bits` at `message.rs:529`.

Reply:
We're using `hubpack::serialize` on the SP size, to avoid thinking about endianness, so I'll remove the comment in `message.rs`.

Reply:
Yeah, I like the loop better; done in 514b45fbe512a91cc4f0bd68ee4287aa860256e2

Reply:
Sounds good, the `from_bits` call ignored endianness since it's a single byte. I like it.